src/pipeline.hpp

```diff
@@ -39,9 +42,10 @@ class Pipeline {
     const std::string name;
     Node& entry;
     Node& exit;
+    ModelMetricReporter& reporter;
 
 public:
-    Pipeline(Node& entry, Node& exit, const std::string& name = "default_name");
+    Pipeline(Node& entry, Node& exit, ModelMetricReporter& reporter, const std::string& name = "default_name");
```

Comment:
Why do we need to pass it here?

Reply:
We need to pass reporter to pipeline constructor in order to increment `requests_success`/`requests_fail` metric of `PipelineDefinition`.